pennylane/devices/qubit/simulate.py

```diff
@@ -279,9 +279,33 @@ def simulate(
 
     has_mcm = any(isinstance(op, MidMeasureMP) for op in circuit.operations)
     if circuit.shots and has_mcm:
-        return simulate_one_shot_native_mcm(
-            circuit, debugger=debugger, rng=rng, prng_key=prng_key, interface=interface
+        results = []
+        aux_circ = qml.tape.QuantumScript(
+            circuit.operations,
+            circuit.measurements,
+            shots=[1],
+            trainable_params=circuit.trainable_params,
         )
+        keys = jax_random_split(prng_key, num=circuit.shots.total_shots)
+        if qml.math.get_deep_interface(circuit.data) == "jax":
+            # pylint: disable=import-outside-toplevel
+            import jax
+
+            def vv(k):
+                return simulate_one_shot_native_mcm(
+                    aux_circ, debugger=debugger, rng=rng, prng_key=k, interface=interface
+                )
+
+            results = jax.vmap(vv, in_axes=(0,))(keys)
+            results = list(zip(*results))
+        else:
+            for i, _ in enumerate(circuit.shots):
+                results.append(
+                    simulate_one_shot_native_mcm(
+                        aux_circ, debugger=debugger, rng=rng, prng_key=keys[i], interface=interface
```

Comment:
If we already know there's no jax involved at this point, do we even need the `prng_key`? Isn't that basically only for jax-native simulations?

Reply:
Correct, let me change it and see if anything breaks.

Reply:
`interface` here is a config keyword, not necessarily matching the deep interface of the tape data. We allow using JAX's RNG for sampling with NumPy arrays, in which case we should still pass the RNG keys.